### PR TITLE
Media metadata extraction & Thumbnailer rework

### PR DESCRIPTION
**Pull request by HeavenVolkoff**
- Extract media (video/audio) metadata with FFmpeg then save it on db and present on front-end
- Model FFmpeg metadata to database
- Rename media_data to exif_metadata
- Integrate FFmpeg metadata extraction into job system and watcher
- Fix thumbnails generation being broken for portrait videos
- Greatly improve thumbnailer logic
- Integrate ffmpeg and media_metadata crate
- Remove some unused and commented out code
- Rework, improve logic and rename byte_size to humanize_size

### CODE REVIEW

File: crates/ffmpeg/Cargo.toml [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
 version = "0.1.0"
-authors = ["Ericson Soares <ericson.ds999@gmail.com>"]
+authors = [
+	"Ericson Soares <ericson@spacedrive.com>",
+	"Vítor Vasconcellos <vitor@spacedrive.com>",
+]
 readme = "README.md"
 description = "A simple library to generate video thumbnails using ffmpeg with the webp format"
-rust-version = "1.64.0"
+rust-version = "1.77"
```
Comment (fogodev) [inline crates/ffmpeg/Cargo.toml:10]: The same I mentioned about core's `Cargo.toml`

File: crates/prisma/src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#![recursion_limit = "256"]
```
Comment (fogodev) [inline crates/prisma/src/lib.rs:1]: Same as the other ones

**Review by fogodev — CHANGES_REQUESTED** [review]

File: core/crates/prisma-helpers/src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#![recursion_limit = "256"]
```
[the remark this answers is not in the document]
Comment (HeavenVolkoff) [reply core/crates/prisma-helpers/src/lib.rs:1]: Yes, running tests and building the server breaks if these higher recursion limits are not defined

File: core/src/lib.rs [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#![recursion_limit = "256"]
```
[the remark this answers is not in the document]
Comment (HeavenVolkoff) [reply core/src/lib.rs:1]: Yes, running tests and building the server breaks if these higher recursion limits are not defined

File: core/crates/heavy-lifting/src/job_system/job.rs [hunk 5]
```diff
@@ -29,6 +29,7 @@ use tokio::{
 	spawn,
 	sync::{watch, Mutex},
 };
+use tokio_stream::StreamExt;
```
[the remark this answers is not in the document]
Comment (HeavenVolkoff) [reply core/crates/heavy-lifting/src/job_system/job.rs:32]: Probably a merge mistake, will revert

File: crates/prisma/src/lib.rs (hunk 2 above)
Comment (HeavenVolkoff) [reply crates/prisma/src/lib.rs:1]: Yes, running tests and building the server breaks if these higher recursion limits are not defined

**Review by fogodev — APPROVED** [review]

**Review by fogodev — APPROVED** [review]
LGTM

**Pull request merged**